contracts/oracle/MarketGovernanceOracle.sol

```diff
@@ -0,0 +1,338 @@
+        uint256 _actualChangeRate,
+        uint256 _liquidityJumpTarget,
+        uint256 _periodStartTime,
+        address _oracle
+    ) CoreRefV2(_core) {
+        baseChangeRate = _baseChangeRate;
+        actualChangeRate = _actualChangeRate;
+        liquidityJumpTarget = _liquidityJumpTarget;
+        periodStartTime = _periodStartTime;
+        oraclePrice = IVoltSystemOracle(_oracle).getCurrentOraclePrice();
+    }
+
+    /// @notice only callable by the pcv oracle smart contract
+    modifier onlyPCVOracle() {
+        require(msg.sender == pcvOracle, "MGO: Not PCV Oracle");
+        _;
+    }
+
+    // ----------- Getters -----------
+
+    /// @notice get the current scaled oracle price
+    /// applies the change rate smoothly over a 30.42 day period
+    /// scaled by 18 decimals
+    // prettier-ignore
+    function getCurrentOraclePrice() public view returns (uint256) {
+        uint256 cachedStartTime = periodStartTime; /// save a single warm SLOAD if condition is false
+        if (cachedStartTime >= block.timestamp) { /// only accrue interest after start time
+            return oraclePrice;
+        }
+
+        uint256 cachedOraclePrice = oraclePrice; /// save a single warm SLOAD by using the stack
+        uint256 timeDelta = Math.min(block.timestamp - cachedStartTime, TIMEFRAME);
+        /// todo validate this works the same as VoltSystemOracle in fuzz, unit and invariant tests
+        uint256 periodPriceChange = cachedOraclePrice * actualChangeRate;
+        uint256 priceDelta = periodPriceChange * timeDelta / TIMEFRAME / SCALE;
+
+        return cachedOraclePrice + priceDelta;
+    }
+
+    /// TODO verify this calculation with an invariant test, a differential test, a unit test, and a fuzz test
+    /// @notice function used to calculate the new actual rate based on liquidity
+    /// @param liquidPercentage is the percent of total PCV that is liquid
+    /// @return the new actual rate based on liquidity profile in the Volt system
+    function calculateRate(uint256 liquidPercentage)
+        public
+        view
+        returns (uint256)
+    {
+        uint256 _liquidityJumpTarget = liquidityJumpTarget; /// save 2 warm SLOADs
+        /// liquidity jump target is the lowest percentage liquid before Volt rate spikes
+        /// if more liquid than jump target, just use base rate
+        /// if less liquid than jump target, boost Volt rate
+        /// max change rate is base rate, return base rate
+        if (
+            liquidPercentage >= liquidityJumpTarget ||
+            maximumChangeRate == baseChangeRate
+        ) {
+            return baseChangeRate;
+        }
+
+        /// use LERP to discover PCM delta on top of base rate
+        uint256 totalPossibleBoost = maximumChangeRate - baseChangeRate;
+
+        /// example figures
+
+        /// pcm boost can be between 0 and 1000,
+        /// with 1000 being 100 basis points or 1% on top of the base rate
+        /// totalPossibleBoost = 1000 pcm
```

Comment:
we can remove this comment as it uses PCM and we're using PPQ (parts per quintrillion)